### PR TITLE
kie-issues#250: Width changes are not correctly persisted

### LINKED ISSUE #250: Width changes are not correctly persisted
This is a ticket to map all the persistence issues found during the tests in the new Boxed Expression Editor.

**Issue 1: List widths and Business Knowledge Models (Function expressions) widths are not persisted**

1. Add a List expression
2. Resize its size
3. Save the DMN file
4. Open it again
5. You will notice that the size is not persisted

**Issue 2: Annotation column in inner Decision Tables**

1. Add a Context Expression
2. Add an inner Decision Table
3. Resize the annotation column
4. Save the DMN
5. Open it again
6. The size of the Annotation Column is not persisted

```[tasklist]
### Acceptance Criteria
- [x] Literal: Width changes should be persisted (modify - save - reopen the dmn file)
- [x] Context: Width changes should be persisted (modify - save - reopen the dmn file)
- [x] Relation: Width changes should be persisted (modify - save - reopen the dmn file)
- [x] List: Width changes should be persisted (modify - save - reopen the dmn file)
- [x] Function: Width changes should be persisted (modify - save - reopen the dmn file)
- [x] Invocation: Width changes should be persisted (modify - save - reopen the dmn file)
- [x] Decision Table: Width changes should be persisted (modify - save - reopen the dmn file)
- [x] Nested Literal: Width changes should be persisted (modify - save - reopen the dmn file)
- [x] Nested Context: Width changes should be persisted (modify - save - reopen the dmn file)
- [x] Nested Relation: Width changes should be persisted (modify - save - reopen the dmn file)
- [x] Nested List: Width changes should be persisted (modify - save - reopen the dmn file)
- [x] Nested Function: Width changes should be persisted (modify - save - reopen the dmn file)
- [x] Nested Invocation: Width changes should be persisted (modify - save - reopen the dmn file)
- [x] Nested Decision Table: Width changes should be persisted (modify - save - reopen the dmn file)
```

=======

Original issues (moved to other tickets)

**Issue 1: The name of "output columns" if a single output column exist is not persisted**

1. Add a decision table with a single output column
image

2. Save the DMN file
3. Open it again
image

4. You will notice that the "output-1" now is called "Decision-1"

This issue only happens if there is only a single output column.

**Issue 4: Empty expressions width**

Currently user can resize columns with empty expressions (the "Select Expression" cell). But there is nowhere to save the width information since the expression in this case is null.
User should not be allowed to resize those columns.

### PR DESCRIPTION
**Pull request by yesamer**
Closes https://github.com/kiegroup/kie-issues/issues/250

The aim of this PR is to correctly persist any width change on BEE in case of Nested Expression.

To correctly persist a column with a nested expression, it's necessary to calculate the sum of all columns that compose the nested expression. The logic I wrote determines that sum for all expression types cases, and it's recursive: that means it should support multiple levels of nested expressions.

### CODE REVIEW

File: packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/jsinterop/util/ExpressionModelFiller.java [hunk 1]
```diff
@@ -132,38 +149,71 @@ public static void fillListExpression(final List listExpression,
...
     }
 
+    private static Double retrieveFunctionExpressionWidth(FunctionProps props) {
+        Double functionExpressionWidth = null;
+        if (props.functionKind.equals(FunctionKind.FEEL.value())) {
+            functionExpressionWidth = retrieveNestedExpressionWidth(((FeelFunctionProps) props).expression);
+        } else if (props.functionKind.equals(FunctionKind.JAVA.value())) {
+            functionExpressionWidth = props.classAndMethodNamesWidth;
+        } else if (props.functionKind.equals(FunctionKind.PMML.value())) {
+            functionExpressionWidth = props.classAndMethodNamesWidth;
```
Comment (jomarko) [inline packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/jsinterop/util/ExpressionModelFiller.java]: model and document shouldn't be on this line?

**Review by jomarko — COMMENTED** [review]
One question from me. Thank you for a PR @yesamer

File: packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/jsinterop/util/ExpressionModelFiller.java (hunk 1 above)
Comment (yesamer) [reply packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/jsinterop/util/ExpressionModelFiller.java]: @jomarko Good point, that is a particular scenario. Basically, PMML Functions can't be resized, therefore I put there the only reasonable width value, even is always null in that case. I guess it is better to force that, setting NULL than using an unrelated field.

**Comment by jomarko** [comment]
Thank you @yesamer for updates. I checked built VSIX. One interesting finding. For BKM node is the PMML different than for being nested as context entry. Please notice cell is 'tight' for leading cells in BKM.

### BKM
Screenshot 2023-05-12 072935

### Decision, Context Entry, PMML Function
Screenshot 2023-05-12 073034

**Comment by yesamer** [comment]
> Thank you @yesamer for updates. I checked built VSIX. One interesting finding. For BKM node is the PMML different than for being nested as context entry. Please notice cell is 'tight' for leading cells in BKM.
> ### BKM
> 
> Screenshot 2023-05-12 072935
> ### Decision, Context Entry, PMML Function
> 
> Screenshot 2023-05-12 073034

@jomarko Thank you, I updated the width, setting the same used in Java Function (140)

**Review by jomarko — APPROVED** [review]
Thank you for updates.

**Pull request merged**